### DISCUSSION TITLE
Dependency array not causing re-render in derived atom with getter arguments

### DISCUSSION THREAD

**Discussion by sdeleur-sc** [pmndrs/jotai#2005] [Bug report] [answered]
When using a derived atom that has an argument that is used in the `read` function, the components that use the derived atom do not get updated when the original atom gets updated

Uncommenting line 19 fixes the issue because it introduces a direct dependency in the component:
https://codesandbox.io/s/jotai-render-issue-2nkwrw?file=/src/App.jsx

**Comment by dai-shi** [comment, marked as answer]
Dependency tracking only works in sync (or async for promise value), so you can't return a getter function as a value.
Here's a canonical solution: https://codesandbox.io/s/jotai-render-issue-forked-x36x44?file=/src/App.jsx

The use case is different, but you might be interested in how [jotai-game](https://github.com/jotaijs/jotai-game) implements `useTransientAtom`.

**Comment by sdeleur-sc** [comment]
got it, thanks for clarifying the behavior!
